cmd/bedMinimumDistanceName/testdata/genomeBed3.bed

```diff
@@ -0,0 +1,6 @@
+chr1	30	40	gene1	0	+
+chr2	1	13	gene2	0	+
+chr2	2	4	gene3	0	-
+chr1	20	30	gene1	0	+
+chr1	30	56	gene4	0	-
+chr1	55	59	gene4	0	-
```

Comment:
From my interpretation, it seems the workflow is to figure out the closest TSS then detect where your input bed is relative to the position of the TSS. For test 3 it seems like the error here is the result of bed.MinimumDistance returning 0 for both TSS on line 5 and 6, so the program defaults to using the first bed entry which shows the input bed as being upstream whereas it is downstream to the TSS on line 6. Maybe this could be resolved by merging overlapping beds before you calculate distance?

Reply:
I wonder how this will look in practice with Chelsea's tandem repeats and gene transcripts. I imagine there will only be 1 gene transcript for each gene to compare to each tandem repeat? If there are multiple gene transcripts for each gene, then as long as the Name field specifies the transcript variant, then there won't be ambiguous entries, so there might not be a need to merge in practice?

Reply:
@dasnellings Close on the workflow. We are not looking genome wide for the closest TSS, but matching on the Name field of the two beds, and then determine the minimum distance between them. I will be inputing a genome wide bed of all genes with the coordinates being only the TSS of each gene. 
Great catch! You point out something in that I didnt quite account for which is there being matching genes where the TR ends up being in one instance upstream and another downstream. I wrote in this case because of my realization that you can have many ensembl transcripts for one ensembl gene, and I liked the idea of being able to handle them both. However, I do not anticipate needing this for how I've already pulled the data for my TSS genomeBed. So I think it will be best to remove this case from my test file and make a note in my code that all Names in the genomeBed must be unique.

Reply:
@RavenYL Yes! You are correct. There does appear to be only one transcript whose expression matches to any given eSTR. But the naming convention is neither normal ensembl nor UCSC so I was working through an odd case that I'm now thinking isnt worth my time. 

I'm going to remove this case from my code and make sure my usage statement covers this expectation. 

Thank you both for your help!